projects/core/components/data-list/data-list.component.ts

```diff
@@ -115,6 +115,8 @@ export class TuiDataListComponent<T>
     }
 
     private get elements(): readonly HTMLElement[] {
-        return Array.from(this.el.querySelectorAll('a,button,input'));
+        return Array.from(
+            this.el.querySelectorAll('a[tuiOption],button[tuiOption],input[tuiOption]'),
```

Comment:
I have doubts about `input[tuiOption]` 🤔 
Should we leave it as it was ?

https://github.com/taiga-family/taiga-ui/blob/b5cafe5cea5f761a1183c3764b334723b95609c8/projects/core/components/data-list/option.component.ts#L25